packages/react-hooks/src/hooks/use-cart/use-cart.reducer.ts

```diff
@@ -55,48 +41,58 @@ const cartReducer = (
           ? action.payload.variant.id
           : action.payload.id;
 
+      const cart: CartItem[] = state.cart.filter((item) => item.id !== payloadId);
+
+      state.setCacheItem('cart', JSON.stringify(cart));
       return {
         ...state,
-        cart: state.cart.filter((item) => item.id !== payloadId)
+        cart
       };
     }
+    case INCREMENT_ITEM: {
+      const cart: CartItem[] = state.cart.map((item) => {
+        const payloadId =
+          'variant' in action.payload
+            ? action.payload.variant.id
+            : action.payload.id;
+
+        if (item.id === payloadId) {
+          return { ...item, quantity: item.quantity + 1 };
+        }
+
+        return item;
+      });
 
-    case INCREMENT_ITEM:
+      state.setCacheItem('cart', JSON.stringify(cart));
       return {
         ...state,
-        cart: state.cart.map((item) => {
-          const payloadId =
-            'variant' in action.payload
-              ? action.payload.variant.id
-              : action.payload.id;
-
-          if (item.id === payloadId) {
-            return { ...item, quantity: item.quantity + 1 };
-          }
-
-          return item;
-        })
+        cart
       };
+    }
     case DECREMENT_ITEM:
+      const cart: CartItem[] = state.cart.map((item) => {
+        const payloadId =
+          'variant' in action.payload
+            ? action.payload.variant.id
+            : action.payload.id;
+
+        if (item.id === payloadId) {
+          return {
+            ...item,
+            quantity: item.quantity >= 1 ? item.quantity - 1 : item.quantity
+          };
+        }
+
+        return item;
+      })
+
+      state.setCacheItem('cart', JSON.stringify(cart));
       return {
         ...state,
-        cart: state.cart.map((item) => {
-          const payloadId =
-            'variant' in action.payload
-              ? action.payload.variant.id
-              : action.payload.id;
-
-          if (item.id === payloadId) {
-            return {
-              ...item,
-              quantity: item.quantity >= 1 ? item.quantity - 1 : item.quantity
-            };
-          }
-
-          return item;
-        })
+        cart
       };
     case CLEAR_CART:
+      window.localStorage.removeItem('cart')
```

Comment:
Similar to `setItem` -- does this need to be accounted for if there is no `window`?